src/index.js

```diff
@@ -0,0 +1,180 @@
+                params: { lat: lat, lon: lon }
+            });
+        })
+        .then(weatherResponse => {
+            const tempKelvin = weatherResponse.data.main.temp;
+            const tempFahrenheit = Math.round((tempKelvin - 273.15) * 9 / 5 + 32);
+            document.getElementById('tempValue').textContent = tempFahrenheit;
+
+            changeTempColors();
+            changeLandscapes();
+        })
+        .catch(error => {
+            console.error('Error fetching weather:', error);
+            document.getElementById('tempValue').textContent = 'N/A';
+        });
+};
+
+
+// Wave 5: Selecting the Sky
+const updateSky = () => {
+    const sky = document.getElementById("skySelect").value;
+    const skyDisplay = document.getElementById("sky");
+    const gardenContent = document.getElementById("gardenContent");
+
+    const skyOptions = {
+        sunny: "☁️ ☁️ ☁️ ☀️ ☁️ ☁️",
+        cloudy: "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️",
+        rainy: "🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧",
+        snowy: "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨",
+    };
+
+    skyDisplay.textContent = skyOptions[sky];
+    gardenContent.classList.remove('sunny', 'cloudy', 'rainy', 'snowy');
+    gardenContent.classList.add(sky);
+};
+
+
+// Wave 6: Resetting the City Name
+const resetCityName = () => {
+    state.cityNameInput.value = 'Seattle';
+    state.headerCityName.textContent = state.cityNameInput.value;
+}
+
+
+const registerEvents = () => {
+    state.increaseTempControl.addEventListener('click', handleIncreaseTempControlClicked);
+    state.decreaseTempControl.addEventListener('click', handleDecreaseTempControlClicked);
+    state.cityNameInput.addEventListener('input', changeHeaderCityName);
+    state.cityNameReset.addEventListener('click', resetCityName);
+    state.currentTempButton.addEventListener('click', getWeatherForCity);
+    state.skySelect.addEventListener('change', updateSky);
+}
+
+
+const onloaded = () => {
+    loadedControls();
+    registerEvents();
+    changeTempColors();
+    changeLandscapes();
+    changeHeaderCityName();
+    updateSky();
+}
+
+
+const loadedControls = () => {
+    // Wave 2
+    state.tempValue = document.getElementById('tempValue');
+    state.increaseTempControl = document.getElementById('increaseTempControl');
+    state.decreaseTempControl = document.getElementById('decreaseTempControl');
+    state.landscape = document.getElementById('landscape');
+
+    // Wave 3
+    state.cityNameInput = document.getElementById('cityNameInput');
+    if (!state.cityNameInput.value) {
+        state.cityNameInput.value = 'Seattle';
+    }
+
+    state.headerCityName = document.getElementById('headerCityName');
+
+    // Wave 4
+    state.currentTempButton = document.getElementById('currentTempButton');
+
+    // Wave 5
+    state.skySelect = document.getElementById('skySelect');
+
+    // Wave 6
+    state.cityNameReset = document.getElementById('cityNameReset');
+}
+
+
+onloaded();
```

Comment:
We aren't listening for the `DOMContentLoaded` event before looking up elements and attaching event listeners, what are potential issues that could arise due to this?